services/management/ia/alerts_service.go

```diff
@@ -18,22 +18,141 @@
 package ia
 
 import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/percona-platform/saas/pkg/common"
+	"github.com/percona/pmm/api/alertmanager/ammodels"
+	"github.com/percona/pmm/api/managementpb"
 	iav1beta1 "github.com/percona/pmm/api/managementpb/ia"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
 )
 
+// AlertsService represents integrated alerting alerts API.
 type AlertsService struct {
-	iav1beta1.UnimplementedAlertsServer // TODO remove
-
-	db *reform.DB
+	db               *reform.DB
+	l                *logrus.Entry
+	alertManager     alertManager
+	templatesService *TemplatesService
 }
 
-func NewAlertsService(db *reform.DB) *AlertsService {
+// NewAlertsService creates new alerts API service.
+func NewAlertsService(db *reform.DB, alertManager alertManager, templatesService *TemplatesService) *AlertsService {
 	return &AlertsService{
-		db: db,
+		l:                logrus.WithField("component", "management/ia/alerts"),
+		db:               db,
+		alertManager:     alertManager,
+		templatesService: templatesService,
 	}
 }
 
+// ListAlerts returns list of existing alerts.
+func (s *AlertsService) ListAlerts(ctx context.Context, req *iav1beta1.ListAlertsRequest) (*iav1beta1.ListAlertsResponse, error) {
```

Comment:
please add the condition to check the settings to see if IA is enabled otherwise return an error, for all the API methods

Reply:
Done